include/ruby/internal/gc.h

```diff
@@ -417,4 +421,422 @@ void rb_gc_register_mark_object(VALUE object);
 
 RBIMPL_SYMBOL_EXPORT_END()
 
+/**
+ * @private
+ *
+ * @deprecated  This macro once was a thing in the old days, but makes no sense
+ *              any  longer today.   Exists  here  for backwards  compatibility
+ *              only.  You can safely forget about it.
+ */
+#undef USE_RGENGC
+#define USE_RGENGC 1
+
+/**
+ * @private
+ *
+ * This is  a compile-time flag  to enable/disable incremental GC  feature.  It
+ * has to  be set at  the time  ruby itself compiles.   Makes no sense  for 3rd
+ * parties.  It is safe  for them to set this though;  that just doesn't change
+ * anything.
+ */
+#ifndef USE_RINCGC
+# define USE_RINCGC 1
+#endif
+
+/**
+ * @deprecated  This macro seems  broken.  Setting this to  anything other than
+ *              zero just doesn't compile.  We need to KonMari.
+ */
+#ifndef USE_RGENGC_LOGGING_WB_UNPROTECT
+# define USE_RGENGC_LOGGING_WB_UNPROTECT 0
+#endif
+
+/**
+ * @private
+ *
+ * This  is   a  compile-time   flag  to   enable/disable  write   barrier  for
+ * struct ::RArray.  It has to be set  at the time ruby itself compiles.  Makes
+ * no sense for 3rd parties.
+ */
+#ifndef RGENGC_WB_PROTECTED_ARRAY
+# define RGENGC_WB_PROTECTED_ARRAY 1
```

Comment:
BTW these kind of flags are for debugging RGenGC (*1), so I think we can remove them...

*1: try disable RGenGC per type if we found an issue around array.